### PR TITLE
Addressing Issue #25

### PR DESCRIPTION
**Pull request by adebellis**
Hi,
I extended the scope of the changes to address both number and string literals (as suggested by the todo).
I opted for a general error message that can also be used for other kinds of literal (i.e. ({ true })) is not valid).
I'll wait for your feedback!

### CODE REVIEW

File: src/quick-lint-js/error.h [hunk 1]
```diff
@@ -246,6 +246,9 @@
       .error(u8"variable used before declaration: {0}", use)                   \
           .note(u8"variable declared here", declaration))                      \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      invalid_lone_literal_in_object_literal, { source_code_span where; },     \
+      .error(u8"invalid lone literal in object literal", where))               \
```
Comment (strager) [inline src/quick-lint-js/error.h:251]: Nit: I'd like to keep error messages in sorted order (sorted by the error type name).

File: src/quick-lint-js/error.h [hunk 2]
```diff
@@ -246,6 +246,9 @@
       .error(u8"variable used before declaration: {0}", use)                   \
           .note(u8"variable declared here", declaration))                      \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      invalid_lone_literal_in_object_literal, { source_code_span where; },     \
```
Comment (strager) [inline src/quick-lint-js/error.h:250]: The name of error types should start with `error_`.
```suggestion
      error_invalid_lone_literal_in_object_literal, { source_code_span where; },     \
```

File: src/parse.cpp [hunk 3]
```diff
@@ -717,11 +717,17 @@ expression_ptr parser::parse_object_literal() {
...
+        if (key_type == token_type::string || key_type == token_type::number) {
+          expression_ptr value = this->make_expression<expression::_invalid>();
+          this->error_reporter_->report(
+              invalid_lone_literal_in_object_literal{key_span});
+          entries.emplace_back(key, value);
+        } else {
+          expression_ptr value = this->make_expression<expression::variable>(
+              identifier(key_span), key_type);
+          entries.emplace_back(key, value);
+        }
```
Comment (strager) [inline src/parse.cpp:730]: :+1:

**Review by strager — APPROVED** [review]
Looks great, aside from some minor comments. Let me know when I should merge your changes.

**Comment by adebellis** [comment]
Sorry for some possibly redundant commits.
You can go ahead and merge if everything is good!

**Comment by strager** [comment]
Landed as commit 502e60ca22cacf9f77a0d51170856efcea205576.

**Pull request closed without merging**